activerecord/lib/active_record/reflection.rb

```diff
@@ -516,6 +516,8 @@ def initialize(name, scope, options, active_record)
         @foreign_key = nil
         @association_foreign_key = nil
         @association_primary_key = nil
+        # setting foreign key to an Array should behave exactly like setting `query_constraints`
+        options[:query_constraints] = options.delete(:foreign_key) if options[:foreign_key].is_a?(Array)
```

Comment:
Can you multiline this for readability and update the comment? I found the existing comment not super clear.

```suggestion
        # If the foreign key is an array, set query constraints options and don't use the foreign key
        if options[:foreign_key].is_a?(Array)
          options[:query_constraints] = options.delete(:foreign_key)
        end
```

Reply:
Done!